File: python/with_java/programmers/level 1/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for n in range(left, right+1):
        cnt = 0
        for s in range(1, int(n ** 0.5)+1):
            if n%s != 0:
                continue
            cnt += 1
            if n/s != s:
                cnt += 1
        if cnt % 2 == 0:
            answer += n
        else:
            answer -= n
    return answer

File: java/src/main/java/ps_traning/programmers/level_1/약수의_개수와_덧셈.java
package ps_traning.programmers.level_1;

public class 약수의_개수와_덧셈 {
    public int solution(int left, int right) {
        int answer = 0;
        for (int n = left; n <= right; n++) {
            int cnt = 0;
            for (int s = 1; s <= (int) Math.sqrt(n); s++) {
                if (n % s != 0) {
                    continue;
                }
                cnt += 1;
                if (n / s != s) {
                    cnt += 1;
                }
            }
            if (cnt % 2 == 0) {
                answer += n;
            } else {
                answer -= n;
            }
        }
        return answer;
    }
}
